account_invoice_report_hls/report/report_invoice_delivery_templates.xml

```diff
@@ -131,8 +131,15 @@
                                 </tr>
                             </thead>
                             <tbody class="invoice_tbody">
-                                <tr t-foreach="o.line_ids" t-as="l">
-                                    <td><span t-field="l.date_delivered" t-options='{"format": "yyyy-MM-dd"}'/></td>
+                                <tr t-foreach="o._get_sorted_line_ids()" t-as="l">
+                                    <td>
+                                        <t t-if="l.date_delivered">
+                                            <span t-field="l.date_delivered" t-options='{"format": "yyyy-MM-dd"}'/>
+                                        </t>
+                                        <t t-elif="l.sale_line_id and l.sale_line_id.order_id.commitment_date">
+                                            <span t-field="l.sale_line_id.order_id.commitment_date" t-options='{"format": "yyyy-MM-dd"}'/>
+                                        </t>
+                                    </td>
```

Comment:
I guess we don't need this. Can we instead just assign `date_delivered` when report line is created?